### PR TITLE
Refactor code, add more curse words, add `chat_anticurse.is_curse_found` function for other mods to use, and remove mod loading warnings.

### PR DESCRIPTION
**Pull request by livphi**
I refactored the code, made it easier to read. And I also added more curse words from another (now deleted) mod. I made an API for other mods to use (just one function), so this mod won't conflict with the ranks mod. And removed the description.txt and depends.txt files and moved them all into mod.conf.

### CODE REVIEW

**Comment by AndrejIT** [comment]
Hello! Nice to see life here, thank you for contributing!
Any information why chat_filter mod got deleted?
I am not comfortable publicly placing curse words in plaintext, but let's see what we can do...

File: init.lua [hunk 1]
```diff
@@ -1,131 +1,207 @@
...
+
+	local old_command = minetest.chatcommands["me"].func
+
+	minetest.chatcommands["me"].func = function(name, param)
+
+	if chat_anticurse.is_curse_found(name, param) then
+		return nil
+	end
+
+	return old_command(name, param)
```
Comment (Niklp09) [inline init.lua]: Missing indentation

File: init.lua [hunk 2]
```diff
@@ -1,131 +1,207 @@
...
+	
+	local old_command = minetest.chatcommands["msg"].func
+	
+	minetest.chatcommands["msg"].func = function(name, param)
+	
+	if chat_anticurse.is_curse_found(name, param) then
+		return nil
+	end
+
+	return old_command(name, param)
```
Comment (Niklp09) [inline init.lua]: Missing indentation

File: init.lua [hunk 3]
```diff
@@ -1,131 +1,207 @@
...
+	" scheiße ",
+}
+
+-- Returns true if a curse word is found
+function chat_anticurse.is_curse_found(name, message)
+
+	-- Is nil when minetest loads i guess?
+	if not message then
+		return false
+	end
```
Comment (Niklp09) [inline init.lua]: Is this really needed?

File: mod.conf [hunk 4]
```diff
@@ -1 +1,4 @@
 name = chat_anticurse
+description = Kick player out when they type curse word in chat or using /me or /msg command. Checks for some English, Russian, and German curse words.
+optional_depends = default
```
Comment (Niklp09) [inline mod.conf]: `default` isn't used anywhere in this mod, pretty sure this dependency is useless.

Comment (livphi) [reply mod.conf]: The original mod author had `default` as an optional dependency, im not sure why?

**Comment by livphi** [comment]
> Hello! Nice to see life here, thank you for contributing! Any information why chat_filter mod got deleted? I am not comfortable publicly placing curse words in plaintext, but let's see what we can do...

I can change it to not have the words in plaintext. I only did it because thats how chat_filter stored the curse words. I'm not sure why they deleted the mod? Its on ContentDB https://content.minetest.net/packages/xXNicoXx/chat_filter/ but the source code is missing from their profile? https://github.com/NoNameDude/minetest-chat-filter https://github.com/NoNameDude

File: init.lua (hunk 1 above)
Comment (livphi) [reply init.lua]: Fixed indentation

File: init.lua (hunk 3 above)
Comment (livphi) [reply init.lua]: Removed

File: mod.conf (hunk 4 above)
Comment (livphi) [reply mod.conf]: Tested the mod in devtest, default is not needed

**Comment by livphi** [comment]
@AndrejIT i obfuscated the curse words. Is it ready to merge?

File: init.lua [hunk 5]
```diff
@@ -1,131 +1,217 @@
...
-    if uncensored == 2 then
-        minetest.kick_player(name, "Cursing or words, inappropriate to game server. Kids may be playing here!")
-        minetest.chat_send_all("Player <"..name.."> warned for cursing" )
-        minetest.log("action", "Player "..name.." warned for cursing. Chat:"..message)
-        return true
-    end
+-- Returns true if a curse word is found
+function chat_anticurse.is_curse_found(name, message)
 
+	local checkingmessage=string.lower( name.." "..message .." " )
```
Comment (Niklp09) [inline init.lua]: Codesyle
```suggestion
	local checkingmessage = string.lower( name.." "..message .." " )
```

**Review by Niklp09 — APPROVED** [review]
LGTM, please address my last comment. Didn't check the obfuscated bad word list.

File: init.lua (hunk 5 above)
Comment (livphi) [reply init.lua]: This was how the original author typed the code. Fixed.

**Comment by livphi** [comment]
It should be ready to merge now. It now checks for words with asterisks so people can't curse like that (if you want to include those commits).

**Pull request merged**